raft.go

```diff
@@ -1650,51 +1869,43 @@ func (r *Raft) electSelf() <-chan *voteResult {
 	}
 
 	// Construct a function to ask for a vote
-	askPeer := func(peer string) {
+	askPeer := func(peer Server) {
 		r.goFunc(func() {
 			defer metrics.MeasureSince([]string{"raft", "candidate", "electSelf"}, time.Now())
-			resp := &voteResult{voter: peer}
-			err := r.trans.RequestVote(peer, req, &resp.RequestVoteResponse)
+			resp := &voteResult{voterGUID: peer.GUID}
+			err := r.trans.RequestVote(peer.Address, req, &resp.RequestVoteResponse)
 			if err != nil {
 				r.logger.Printf("[ERR] raft: Failed to make RequestVote RPC to %v: %v", peer, err)
 				resp.Term = req.Term
 				resp.Granted = false
 			}
-
-			// If we are not a peer, we could have been removed but failed
-			// to receive the log message. OR it could mean an improperly configured
-			// cluster. Either way, we should warn
-			if err == nil {
-				peerSet := decodePeers(resp.Peers, r.trans)
-				if !PeerContained(peerSet, r.localAddr) {
-					r.logger.Printf("[WARN] raft: Remote peer %v does not have local node %v as a peer",
```

Comment:
True - if there are normal cases that can lead to this then a warning isn't appropriate. I'm mostly concerned about knowing if, in the steady state outside of configuration changes, there's ever a discrepancy between the configuration among all members in the cluster. Maybe this would be better done with a running checksum over the applied logs or something like that. This is probably something to think about separate from this PR.